src/fr/fmteam/src/eu/kanade/tachiyomi/extension/fr/fmteam/FMTEAM.kt

```diff
@@ -6,114 +6,126 @@ import eu.kanade.tachiyomi.source.model.MangasPage
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val results = json.decodeFromString<FmteamComicDetailPage>(response.body.string())
+
+        return fmTeamComicToSManga(results.comic)
+    }
+
+    override fun getMangaUrl(manga: SManga): String {
+        return "$baseUrl${manga.url}"
     }
 
     // Chapter list
-    override fun chapterListSelector() = ".list .group .element"
-    override fun chapterFromElement(element: Element): SChapter = SChapter.create().apply {
-        setUrlWithoutDomain(element.select(".title a").attr("href"))
-        name = element.select(".title a").attr("title").trim()
-        val date = element.select(".meta_r").text().trim().split(", ")[1]
-        date_upload = if (date === "Aujourd'hui") {
-            dateFormat.format(Date()).toLong()
-        } else {
-            try {
-                dateFormat.parse(date)!!.time
-            } catch (e: ParseException) {
-                0L
-            }
-        }
+    override fun chapterListRequest(manga: SManga): Request = GET("$baseUrl/api${manga.url}")
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val results = json.decodeFromString<FmteamComicDetailPage>(response.body.string())
+
+        return results.comic.chapters?.map(::fmTeamChapterToSChapter) ?: emptyList()
+    }
+
+    override fun getChapterUrl(chapter: SChapter): String {
+        return "$baseUrl${chapter.url}"
     }
 
     // Pages
-    override fun pageListParse(document: Document): List<Page> {
-        val pages = mutableListOf<Page>()
-        val script = document.selectFirst("script:containsData(pages =)")!!.data()
-        val allPages = allPagesRegex.findAll(script)
-        allPages.asIterable().mapIndexed { i, it ->
-            pages.add(Page(i, "", it.groupValues[1].replace("\\/", "/")))
+    override fun pageListRequest(chapter: SChapter): Request = GET("$baseUrl/api${chapter.url}")
+
+    override fun pageListParse(response: Response): List<Page> {
+        val results = json.decodeFromString<FmteamChapterDetailPage>(response.body.string())
+
+        return results.chapter.pages?.mapIndexed {
+                i, page ->
+            Page(i, "${results.chapter.url}#${i + 1}", page)
+        } ?: emptyList()
+    }
+
+    override fun imageUrlParse(response: Response): String = throw UnsupportedOperationException("Not used.")
+
+    // Utils
+    private fun fmTeamComicToSManga(comic: FmteamComic): SManga = SManga.create().apply {
+        url = comic.url
+        title = comic.title
+        artist = comic.artist
+        author = comic.author
+        description = comic.description
+        genre = comic.genres.joinToString { it.name }
+        status = when (comic.status) {
+            "En cours" -> SManga.ONGOING
+            "Termin\u00e9" -> SManga.COMPLETED
+            else -> SManga.UNKNOWN
         }
-        return pages
+        thumbnail_url = comic.thumbnail
```

Comment:
```suggestion
        thumbnail_url = comic.thumbnail
        initialized = true
```
since all the info is already available at popular/latest, mark the `SManga` objects as initialized so as to avoid calling `fetchMangaDetails`, saving a network call

Reply:
I thought that setting the value was enough to avoid the call, my bad.

Reply:
I just want to point out that the documentation isn't very clear on this behaviour in my opinion.

> `SManga.initialized` tells the app if it should call `fetchMangaDetails`. If you are overriding `fetchMangaDetails`, make sure to pass it as `true`.